combined.js

```diff
@@ -415,20 +420,40 @@ function drawCanvas(canvas, img, draw_type)
     }
 }
 
-/*
-    TODO: Base function for TFJS based draws.
-    Should handle BodyPix variations (Overlay/Pixels) and Pose-Animator.
-*/
-async function tensor_draw_pixel(canvas, img)
+async function segmentGenerator(video)
 {
-    try{   
-        if (Animator.tfjs_draw_counter==300)
-            throw ReferenceError
-        const partSegmentation = await Animator.net.segmentMultiPersonParts(img);
-        // The colored part image is an rgb image with a corresponding color from the
-        // rainbow colors for each part at each pixel, and black pixels where there is
-        // no part.
-        const coloredPartImage = bodyPix.toColoredPartMask(partSegmentation);
+    while(Animator.video_on==true && (Animator.draw_type==="tfjs-pixel" || Animator.draw_type==="tfjs-blur")){
+        const segmentationStartTimeMs = performance.now()
+
+        if(Animator.draw_type==="tfjs-pixel")
+            Animator.segment = await Animator.net.segmentMultiPersonParts(video)
+        else
+            Animator.segment = await Animator.net.segmentPerson(video)
+
+        lastFrameInferenceTimeMs = performance.now() - segmentationStartTimeMs;
+        const sleepTimeMs = 3 * this.lastFrameInferenceTimeMs;
+        //  console.log("segment attempted")
+        await sleep(sleepTimeMs);
+    }
+    console.log("Exiting segGen")
+}
+
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+function drawSegment(canvas, img)
```

Comment:
Draw segments continuously, regardless of them being new/stale, with respective draw function based on the Blur/Pixel mode.